packages/scandipwa/src/component/CheckoutAgreements/CheckoutAgreements.component.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * ScandiPWA - Progressive Web App for Magento
+ *
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/base-theme
+ * @link https://github.com/scandipwa/base-theme
+ */
+
+import PropTypes from 'prop-types';
+import { PureComponent } from 'react';
+
+import Field from 'Component/Field';
+
+import './CheckoutAgreements.style.scss';
+
+/**
+ * @class CheckoutAgreements
+ * @namespace Component/CheckoutAgreements/Component
+ */
+export class CheckoutAgreements extends PureComponent {
+    static propTypes = {
+        termsAreEnabled: PropTypes.bool.isRequired,
+        termsAndConditions: PropTypes.array.isRequired,
+        setAgreementsStatus: PropTypes.func.isRequired,
+        showAgreementPopup: PropTypes.func.isRequired
+    };
+
+    state = {
+        isTermsAndConditionsAccepted: false,
+        selectedAgreementsList: [],
+        requiredAgreementsList: []
+    };
+
+    componentDidMount() {
+        const { termsAreEnabled } = this.props;
+
+        if (termsAreEnabled) {
+            this.setRequiredAgreements();
+        } else {
+            this.setState({ isTermsAndConditionsAccepted: true });
+        }
+    }
+
+    setRequiredAgreements = () => {
+        const { termsAndConditions } = this.props;
+        const requiredAgreementsList = [];
+
+        termsAndConditions.map(({ agreement_id }) => requiredAgreementsList.push(agreement_id));
```

Comment:
check out the docs for `map` - it actuall returns an array of the values returned by the function!

so you could do

```
const requiredAgreementsList = termsAndConditions.map(({ agreement_id }) => agreement_id);
```

instead of those 2 lines

Reply:
2 lines => one-line - changed. Good point!

Regarding copy state/props it's neccessery in my opinion. It's needed to keep current state of selected agreement ids and then I can really easy handle update selected agreement list on event click/check on one of them.

E.G initialy I can have 4 agreements (from props) but I unselect 2 and I dont want to go through every agreement. I just to know which one I need to drop or add from my state agreement list.